src/react/useColors.ts

```diff
@@ -0,0 +1,31 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useEffect, useState } from 'react';
+import { AppBridgeBlock } from '../AppBridgeBlock';
+import { FrontifyColor } from '../types/FrontifyColor';
+
+export type UseColorsReturnType = {
+    colorsByPaletteId: FrontifyColor[];
+};
+
+export const useColors = (appBridge: AppBridgeBlock): UseColorsReturnType => {
+    const [colorsByPaletteId, setColorsByPaletteId] = useState<FrontifyColor[]>([]);
+
+    useEffect(() => {
+        const getColorsByColorPaletteId = async (colorPaletteId: number) => {
+            try {
+                const colors = await appBridge.getColorsByColorPaletteId(colorPaletteId);
+
+                setColorsByPaletteId(colors);
+            } catch (error) {
+                console.error('Error: ', error);
+            }
+        };
+
+        getColorsByColorPaletteId(appBridge.getBlockId());
+    }, []);
```

Comment:
Then you can watch for changes 
```suggestion
    }, [colorPaletteId]);
```